pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -391,14 +393,16 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToShootNetworks(ctx context.Cont
 	}
 
 	var shootNetworks []string
-	if v := shoot.Spec.Networking.Nodes; v != nil {
-		shootNetworks = append(shootNetworks, *v)
-	}
-	if v := shoot.Spec.Networking.Pods; v != nil {
-		shootNetworks = append(shootNetworks, *v)
-	}
-	if v := shoot.Spec.Networking.Services; v != nil {
-		shootNetworks = append(shootNetworks, *v)
+	if !shoot.IsWorkerless() {
+		if v := shoot.Spec.Networking.Nodes; v != nil {
+			shootNetworks = append(shootNetworks, *v)
+		}
+		if v := shoot.Spec.Networking.Pods; v != nil {
+			shootNetworks = append(shootNetworks, *v)
+		}
+		if v := shoot.Spec.Networking.Services; v != nil {
```

Comment:
Services can be non nil for workerless shoots. Should this be inside the condition for shoots with workers only?